ADVANCED.md

```diff
@@ -233,8 +233,6 @@ function isPropsForAnchorElement(
   return 'href' in props;
 }
 
-function Clickable(props: ButtonProps): JSX.Element;
-function Clickable(props: AnchorProps): JSX.Element;
```

Comment:
TL;DR: https://palantir.github.io/tslint/rules/unified-signatures/

Overloads are only useful if the input is coupled with the output e.g. 
```ts
function overloaded(input: T): OutputWhenT;
function overloaded(input: U): OutputWhenU;
function overloaded(input: T | U): OutputWhenTOrU {
  if (isT(input)) return doSomethingWithT(input);
  return doSomethingWithU(input);
}

Otherwise we can just use union parameters.

Reply:
It depends on how the two union constituents are related to each other. I've found that if one constituent is a subtype of the other, the type inference breaks in unexpected ways.

Reply:
I'm not sure we want to advocate for both union props and overloaded function component signatures though. These work differently when used to infer props which may be only relevant for library definitions. [Inferring arguments from an overload only considers the first signature](https://github.com/Microsoft/TypeScript/issues/26591).

Maybe it's better to document when we actually need to overload the function signature?

Reply:
I'm fine with leaving it as union, I believe even the TS team encourages union type parameters due to how type unification actually deals with overloads (not very well), so I think it's better 95% of the time.